src/index.js

```diff
@@ -2,41 +2,79 @@
 
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  const price=product.querySelector(".price span").innerHTML;
+  const quantity=product.querySelector(".quantity input").value
+const calculate=price*quantity
+product.querySelector(".subtotal span").innerHTML=calculate
+return calculate
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
+ let productAll=document.getElementsByClassName("product")
+ let count=0
+for(let i=0;i<productAll.length;i++){
 
-  // ITERATION 3
-  //... your code goes here
+  count+=updateSubtotal(productAll[i]);
 }
+let sumTotal=document.querySelector("#total-value span").innerHTML+=count
+;
+return sumTotal
+}
+
 
-// ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
-  //... your code goes here
+
+ let remover=target.parentNode.parentNode
+let parent=target.parentNode.parentNode.parentNode
+parent.removeChild(remover)
 }
 
-// ITERATION 5
 
 function createProduct() {
-  //... your code goes here
+
+  let text=document.querySelectorAll(".create-product")
+  let textDoc=text.querySelector("input")
+  let productName=textDoc.value
+
+let number=document.querySelector(".create-product")
+let textnumber=number.querySelector("input [type='number']")
+let productPrice=textnumber.value
+
+  let productBody=document.createElement(`tr`)
```

Comment:
`productBody.className='product';` here instead of the `productBody.classList.add('product');` below in order to assign the class name 'product' to the table.